.github/workflows/generate_changelog_manual.yml

```diff
@@ -0,0 +1,18 @@
+name: Generate CHANGELOG (Manual)
+on:
+  workflow_dispatch:
```

Comment:
If the only difference is the commit message, I believe these two workflows can be combined. For example, the snapshot  workflow: https://github.com/hashicorp/terraform-provider-aws/blob/f5c1c211773868630fac91a8f47e447ff2c3e398/.github/workflows/snapshot.yml#L3-L6

Reply:
The triggered one does an additional check to see if the PR trigger was merged rather than closed, and the manual one will not have a context of PR, that's why they are separate:

https://github.com/hashicorp/terraform-provider-aws/blob/ad8293cc2ec8556783e4896eed33ae97e56f5919/.github/workflows/generate_changelog.yml#L6-L8

Reply:
Wouldn't this work then? 😄  `if: github.event_name != 'pull_request' || github.event.pull_request.merged`